db/config.cc

```diff
@@ -810,7 +810,7 @@ db::config::config(std::shared_ptr<db::extensions> exts)
             "Disable rewriting streams from cdc_streams_descriptions to cdc_streams_descriptions_v2. Should not be necessary, but the procedure is expensive and prone to failures; this config option is left as a backdoor in case some user requires manual intervention.")
     , strict_allow_filtering(this, "strict_allow_filtering", liveness::LiveUpdate, value_status::Used, strict_allow_filtering_default(), "Match Cassandra in requiring ALLOW FILTERING on slow queries. Can be true, false, or warn. When false, Scylla accepts some slow queries even without ALLOW FILTERING that Cassandra rejects. Warn is same as false, but with warning.")
     , reversed_reads_auto_bypass_cache(this, "reversed_reads_auto_bypass_cache", liveness::LiveUpdate, value_status::Used, true,
-            "Use a new implementation of reversed reads in sstables when performing reversed queries. The new implementation does not require unbounded memory (compared to the old implementation which had to fetch entire partitions into memory) but disables the cache. Turn this option off if your partitions are small so the old implementation is good enough; your queries can then utilize the cache and potentially be faster (since they don't need to use sstables as much). This option is temporary and will be removed as soon as the cache and sstable reverse read algorithms are updated to handle reversed queries correctly.")
+            "Bypass in-memory data cache (the row cache) when performing reversed queries.")
```

Comment:
Right now the `reversed_reads_auto_bypass_cache` option will use the old reversing reader when `false` and perform a regular forward query underneath. This means that the sstable query will be done in forward mode. So this option also gives us the possibility to disable the sstable reverse reading code in case we find that there are bugs in it.

I think we should keep an option for falling back to the old reversing reader for a while.

Anyway, at this point the option is called `reversed_reads_auto_bypass_cache` and the default is `true`. Which would mean that we automatically bypass the cache during reversed queries by default even with this PR?

Reply:
> Right now the `reversed_reads_auto_bypass_cache` option will use the old reversing reader when `false` and perform a regular forward query underneath. This means that the sstable query will be done in forward mode. So this option also gives us the possibility to disable the sstable reverse reading code in case we find that there are bugs in it.
> 
> I think we should keep an option for falling back to the old reversing reader for a while.

I agree that it would be good to have an option for disabling reverse readers globally, which would also avoid memtable reverse readers, not just sstable reverse readers. 

> 
> Anyway, at this point the option is called `reversed_reads_auto_bypass_cache` and the default is `true`. Which would mean that we automatically bypass the cache during reversed queries by default even with this PR?

The default is changed to false in https://github.com/tgrabiec/scylla/commit/65a1a0247a689eb44c0986b7a3cc65fc356a8798